src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicStream.cs

```diff
@@ -51,10 +51,9 @@ private sealed class State
 
             // Buffers to hold during a call to send.
             public MemoryHandle[] BufferArrays = new MemoryHandle[1];
-            public QuicBuffer[] SendQuicBuffers = new QuicBuffer[1];
-
-            // Handle to pinned SendQuicBuffers.
-            public GCHandle SendHandle;
+            public IntPtr SendQuicBuffers  = Marshal.AllocHGlobal(sizeof(QuicBuffer));
```

Comment:
It would be better to allocate this only once it is needed, with the right size.

As written, `SendQuicBuffers` maybe be allocated and never used; or `SendQuicBuffers` may leak if the initialization fails mid-flight.

Reply:
ok. I can move it to constructor at some point or add logic to SendReadOnlyMemoryAsync(). The old code always allocated array of size 1 to optimize for using ReadOnlyMemory e.g. use from HttpClient.